### PR TITLE
[15 min fix] Updated husky package and configuration/pre-commit hook

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We've been ignoring updates to the husky package since they introduced breaking changes. I finally spent a small amount of time updating to the new version.

## QA Instructions, Screenshots, Recordings

1. The only thing to test is to grab this branch locally, change any file that lint-staged runs against, e.g. *.md, *.js etc. and ensure the husky pre-commit hook runs.

For example

```bash
forem on  nickytonline/husky-breaking-changes-upgrade [$+] is 📦 v1.0.0 via  v14.17.3 via 💎 v2.7.2 
❯ g ci "test"
✔ Preparing...
✔ Running tasks...
✔ Applying modifications...
✔ Cleaning up...
[nickytonline/husky-breaking-changes-upgrade 3c1af6b06] test
 1 file changed, 1 insertion(+)
```

### UI accessibility concerns?

N/A

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: N/A
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: This change will be transparent to contributors and core team members.

## [optional] What gif best describes this PR or how it makes you feel?

A husky dancing in a winter coat with pants on

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "test": "test"
   },
   "scripts": {
+    "prepare": "husky install",
```
Comment (nickytonline) [inline package.json:16]: The [npm prepare script](https://docs.npmjs.com/cli/v7/using-npm/scripts) only runs when in the development environment.

File: package.json [hunk 2]
```diff
@@ -27,11 +28,6 @@
     "e2e": "bin/e2e",
     "e2e:noseed": "yarn run e2e --no-seed"
   },
-  "husky": {
```
Comment (nickytonline) [inline package.json:30]: This configuration is no longer required.

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+FILES=$(git diff --cached --name-only --diff-filter=ACMR | sed 's| |\\ |g')
+[ -z "$FILES" ] && exit 0
```
Comment (nickytonline) [inline .husky/pre-commit:3]: Not sure if this is the best way to get staged files. Open to suggestions.

**Comment by nickytonline** [comment]
buildkite is complaining about the node version.

```bash
error pretty-format@27.0.6: The engine "node" is incompatible with this module. Expected version "^10.13.0 \|\| ^12.13.0 \|\| ^14.15.0 \|\| >=15.0.0". Got "14.14.0"
--
  | error Found incompatible module.
```

When buildkite runs, does it not pull from our `.nvmrc` file @djuber or does `ubuntu-latest` currently ship with node 14.14.0? I couldn't tell if it did or not.

https://github.com/forem/forem/blob/15f876bb212ce12805d36528beb3147041d7bac7/.github/workflows/buildkite.yml#L20

File: .husky/pre-commit (hunk 3 above)
Comment (rhymes) [reply .husky/pre-commit:3]: What's `--diff-filter=ACMR` ?
I might be wrong but `git diff --cached --name-only` should be enough

File: .husky/pre-commit [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+
+# Run lint-staged on all staged
+echo "$FILES" | xargs node_modules/.bin/lint-staged
```
Comment (rhymes) [inline .husky/pre-commit:6]: Probably better to invoke it as a yarn command, right?

```suggestion
echo "$FILES" | yarn lint-staged

```

(added a new line which was missing)

**Review by rhymes — COMMENTED** [review]
Works for me! I added a couple of suggestions

File: .github/dependabot.yml [hunk 5]
```diff
@@ -22,11 +22,6 @@ updates:
     schedule:
       interval: "weekly"
     labels: []
-    ignore:
```
Comment (nickytonline) [inline .github/dependabot.yml:25]: No longer required as we've fixed the breaking changes in this PR.

**Review by rhymes — APPROVED** [review]
LGTM!

**Comment by maestromac** [comment]
Build pass :sob: 

But BuildKite is failing because it doesn't have 3.0.2 yet.

**Review by maestromac — APPROVED** [review]
Tested. LGTM!

I vaguely remember that we had to hold off upgrading it because it will no longer be automatically installed pre-commit hook. I see that won't be an issue anymore because it's now using npm test as the hook?

**Comment by nickytonline** [comment]
> Tested. LGTM!
> 
> I vaguely remember that we had to hold off upgrading it because it will no longer be automatically installed pre-commit hook. I see that won't be an issue anymore because it's now using npm test as the hook?

@maestromac, there is still a pre-commit hook, but now you need to do it in a pre-commit script as opposed to a husky configuration in package.json

**Pull request merged**